lib/charms/pgbouncer_operator/v0/pgb.py

```diff
@@ -56,13 +63,26 @@ class PgbConfig(MutableMapping):
     users_section = "users"
     user_types = ["admin_users", "stats_users"]
 
-    def __init__(self, config=None, *args, **kwargs):
+    def __init__(
+        self,
+        config: Union[str, dict, "PgbConfig"] = None,
+        *args,
+        **kwargs,
+    ):
+        """Constructor.
+
+        Args:
+            config: an existing config. Can be passed in as a string, dict, or PgbConfig object.
+                pgb.DEFAULT_CONFIG can be used here as a default dict.
+        """
         self.__dict__.update(*args, **kwargs)
 
         if isinstance(config, str):
             self.read_string(config)
         elif isinstance(config, dict):
             self.read_dict(config)
+        elif isinstance(config, PgbConfig):
+            self.read_dict(config)
```

Comment:
This can be simplified to `isinstance(config, (dict, PgConfig))`.

Reply:
Whoops, missed that. Thanks!